src/main/java/org/altbeacon/beacon/service/ScanJob.java

```diff
@@ -85,13 +83,19 @@ public boolean onStartJob(final JobParameters jobParameters) {
             mStopHandler.postDelayed(new Runnable() {
                 @Override
                 public void run() {
-                    LogManager.i(TAG, "Scan job runtime expired");
+                    LogManager.i(TAG, "Scan job runtime expired: " + ScanJob.this);
                     stopScanning();
                     mScanState.save();
+                    ScanJob.this.jobFinished(jobParameters , false);
 
-                    startPassiveScanIfNeeded();
+                    // need to execute this after the current block or Android stops this job prematurely
+                    mStopHandler.post(new Runnable() {
```

Comment:
This is a bit weird, not so happy about it..  If this doesn't occur after the jobFinished() callback and after the method exits, Android ends up calling  onStopJob() which eventually tries to destroy the CycledLeScanner a second time and throws an ugly, but harmless stack trace because the CycledLeScanner.mScanHandler is already dead.